frontend/static/js/services/authenticationService.js

```diff
@@ -0,0 +1,26 @@
+import ARequestService from "./ARequestService.js";
+import backendURL from "../constants.js";
+
+
+// TODO: Extend for tournaments
+export default class AuthenticationService extends ARequestService {
+	constructor() {
+		super();
+	}
+
+	async postMatch({username, password}) {
+		return super.postRequest(
+			`${backendURL.authenticationURL}match/ `, //${urlParams ? `/${urlParams}` : ''}`,
+			JSON.stringify({
+				username: username,
+				password: password,
+			})
+		);
+	}
+
+	async postAiMatch(URLParams) {
+		const url = new URL(`${backendURL.authenticationURL}match/`);
```

Comment:
meh, i dont think you need another method and a route for the AI match specifically.. otherwise, why not just have a special route for it in the server too? Just adding the parameter separately to either be empty or true / false with the basic Post request should be fine.

Reply:
it is just adding the parameter tokens/match//**?ai_opponent=true** instead of the body with username and password.  I need the postMatch to avoid more code duplication on the tournament flow, because the authentication modals, need to use different urls which need to be constructed in different ways. We can talk about it tomorrow in detail. My next PR might make the reason clearer as well. 

I am not happy with it either, but getting the token in conjunction with authenticating on one flow path, vs simply setting up players on the other, makes this the easier solution imo. But I am open to discuss :D

Reply:
Also, if I were to do the modal flow again, I'd completely redesign it. I coded myself a little into a corner starting with the login and register modals from the very beginning